### PR TITLE
fix task start with file

### LINKED ISSUE #959: Unable to start task with -f flag if the task definition file contains params with no default value
# Version and Operating System

**tkn Version:**
v0.9.0

**Operating System:** 
All

# Expected Behavior
Taskrun should get created

# Actual Behavior
The command is throwing the error `Error: param 'test-param' not present in spec`

# Steps to Reproduce the Problem

1. Create a task definition in a yaml file with params with no default values
2. Start the taskrun using the command `tkn task start -f <yaml file> -p <param name>=<param value>`

# Additional Info
Below are command logs
```
$ cat script-task-python.yaml 
apiVersion: tekton.dev/v1beta1
kind: Task
metadata:
  name: python-script-task-test
spec:
  description: python task
  params:
    - name: test-param
      description: this is test param
  steps:
    - name: step-one
      image: python  
      script: |
        #!/usr/bin/env python3
        print("$(params.test-param)")
        print("Hello from Python!")
        import time
        for i in range(50):
          print("this is task", flush=True)
          time.sleep(2)


$ cat script-task-python.yaml 
apiVersion: tekton.dev/v1beta1
kind: Task
metadata:
  name: python-script-task-test
spec:
  description: python task
  params:
    - name: test-param
      description: this is test param
  steps:
    - name: step-one
      image: python  
      script: |
        #!/usr/bin/env python3
        print("$(params.test-param)")
        print("Hello from Python!")
        import time
        for i in range(50):
          print("this is task", flush=True)
          time.sleep(2)

$ tkn task start -f script-task-python.yaml -p test-param="this is test param"
Error: param 'test-param' not present in spec
[varadhya@localhost testing]$ 
```

### PR DESCRIPTION
**Pull request by pradeepitm12**
Close #959

This fixes task start with file command.

Note: to start a task from file params needs to have its type
speficed, in case of existing task in cluster type is appended by
the controller.

Signed-off-by: Pradeep Kumar <pradkuma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
release-note
```

### CODE REVIEW

File: pkg/task/task.go [hunk 1]
```diff
@@ -129,7 +129,6 @@ func SpecConvertDown(spec *v1beta1.TaskSpec) *v1alpha1.TaskSpec {
 		downTaskSpec.Workspaces = spec.Workspaces
 		downTaskSpec.Results = spec.Results
 		downTaskSpec.Resources = spec.Resources
-		downTaskSpec.Params = spec.Params
```
Comment (pradeepitm12) [inline pkg/task/task.go:132]: v1alpha1 taskspec will have params in `Sepc.Inputs.Params`

File: pkg/taskrun/create.go [hunk 2]
```diff
@@ -36,8 +36,6 @@ func Create(c *cli.Clients, tr *v1beta1.TaskRun, opts metav1.CreateOptions, ns s
 		if err != nil {
 			return nil, err
 		}
-		v1alpha1TaskRun.Kind = "TaskRun"
```
Comment (pradeepitm12) [inline pkg/taskrun/create.go:39]: `Kind` and `APIVersion` is already added in the `ConvertDown` func.

File: pkg/task/task.go [hunk 3]
```diff
@@ -129,7 +129,6 @@ func SpecConvertDown(spec *v1beta1.TaskSpec) *v1alpha1.TaskSpec {
 		downTaskSpec.Workspaces = spec.Workspaces
 		downTaskSpec.Results = spec.Results
 		downTaskSpec.Resources = spec.Resources
```
Comment (piyush-garg) [inline pkg/task/task.go:131]: can you please remove this one also. seems left by mistake

**Comment by pradeepitm12** [comment]
> /lgtm
> 
> We should follow up with an `e2e` test on this for starting a Task with `-f`.
> 
> Is this pr going to include `tkn pipeline start` fixes, or was that not affected by #959?

Thanks for taking the time to review this.
This PR include fix only for #959 
Till now I have not seen any issue with `tkn pipeline start` if I see any issue I will raise a different PR for it.

**Pull request merged**